rosidl_generator_type_description/rosidl_generator_type_description/__init__.py

```diff
@@ -0,0 +1,475 @@
+    'FIELD_TYPE_NESTED_TYPE_BOUNDED_SEQUENCE': 97,
+    'FIELD_TYPE_INT8_BOUNDED_SEQUENCE': 98,
+    'FIELD_TYPE_UINT8_BOUNDED_SEQUENCE': 99,
+    'FIELD_TYPE_INT16_BOUNDED_SEQUENCE': 100,
+    'FIELD_TYPE_UINT16_BOUNDED_SEQUENCE': 101,
+    'FIELD_TYPE_INT32_BOUNDED_SEQUENCE': 102,
+    'FIELD_TYPE_UINT32_BOUNDED_SEQUENCE': 103,
+    'FIELD_TYPE_INT64_BOUNDED_SEQUENCE': 104,
+    'FIELD_TYPE_UINT64_BOUNDED_SEQUENCE': 105,
+    'FIELD_TYPE_FLOAT_BOUNDED_SEQUENCE': 106,
+    'FIELD_TYPE_DOUBLE_BOUNDED_SEQUENCE': 107,
+    'FIELD_TYPE_LONG_DOUBLE_BOUNDED_SEQUENCE': 108,
+    'FIELD_TYPE_CHAR_BOUNDED_SEQUENCE': 109,
+    'FIELD_TYPE_WCHAR_BOUNDED_SEQUENCE': 110,
+    'FIELD_TYPE_BOOLEAN_BOUNDED_SEQUENCE': 111,
+    'FIELD_TYPE_BYTE_BOUNDED_SEQUENCE': 112,
+    'FIELD_TYPE_STRING_BOUNDED_SEQUENCE': 113,
+    'FIELD_TYPE_WSTRING_BOUNDED_SEQUENCE': 114,
+    'FIELD_TYPE_FIXED_STRING_BOUNDED_SEQUENCE': 115,
+    'FIELD_TYPE_FIXED_WSTRING_BOUNDED_SEQUENCE': 116,
+    'FIELD_TYPE_BOUNDED_STRING_BOUNDED_SEQUENCE': 117,
+    'FIELD_TYPE_BOUNDED_WSTRING_BOUNDED_SEQUENCE': 118,
+
+    # Unbounded Sequence Types
+    'FIELD_TYPE_NESTED_TYPE_UNBOUNDED_SEQUENCE': 145,
+    'FIELD_TYPE_INT8_UNBOUNDED_SEQUENCE': 146,
+    'FIELD_TYPE_UINT8_UNBOUNDED_SEQUENCE': 147,
+    'FIELD_TYPE_INT16_UNBOUNDED_SEQUENCE': 148,
+    'FIELD_TYPE_UINT16_UNBOUNDED_SEQUENCE': 149,
+    'FIELD_TYPE_INT32_UNBOUNDED_SEQUENCE': 150,
+    'FIELD_TYPE_UINT32_UNBOUNDED_SEQUENCE': 151,
+    'FIELD_TYPE_INT64_UNBOUNDED_SEQUENCE': 152,
+    'FIELD_TYPE_UINT64_UNBOUNDED_SEQUENCE': 153,
+    'FIELD_TYPE_FLOAT_UNBOUNDED_SEQUENCE': 154,
+    'FIELD_TYPE_DOUBLE_UNBOUNDED_SEQUENCE': 155,
+    'FIELD_TYPE_LONG_DOUBLE_UNBOUNDED_SEQUENCE': 156,
+    'FIELD_TYPE_CHAR_UNBOUNDED_SEQUENCE': 157,
+    'FIELD_TYPE_WCHAR_UNBOUNDED_SEQUENCE': 158,
+    'FIELD_TYPE_BOOLEAN_UNBOUNDED_SEQUENCE': 159,
+    'FIELD_TYPE_BYTE_UNBOUNDED_SEQUENCE': 160,
+    'FIELD_TYPE_STRING_UNBOUNDED_SEQUENCE': 161,
+    'FIELD_TYPE_WSTRING_UNBOUNDED_SEQUENCE': 162,
+    'FIELD_TYPE_FIXED_STRING_UNBOUNDED_SEQUENCE': 163,
+    'FIELD_TYPE_FIXED_WSTRING_UNBOUNDED_SEQUENCE': 164,
+    'FIELD_TYPE_BOUNDED_STRING_UNBOUNDED_SEQUENCE': 165,
+    'FIELD_TYPE_BOUNDED_WSTRING_UNBOUNDED_SEQUENCE': 166,
+}
+
+
+def field_type_type_name(ftype: definition.AbstractType) -> str:
+    value_type = ftype
+    name_suffix = ''
+
+    if isinstance(ftype, definition.AbstractNestedType):
+        value_type = ftype.value_type
+        name_suffix = NESTED_FIELD_TYPE_SUFFIXES[type(ftype)]
+
+    if isinstance(value_type, definition.BasicType):
+        value_type_name = FIELD_VALUE_TYPE_NAMES[value_type.typename]
+    elif isinstance(value_type, definition.AbstractGenericString):
+        value_type_name = FIELD_VALUE_TYPE_NAMES[type(value_type)]
+    elif (
+        isinstance(value_type, definition.NamespacedType) or
+        isinstance(value_type, definition.NamedType)
+    ):
+        value_type_name = 'FIELD_TYPE_NESTED_TYPE'
+
```

Comment:
This is probably handled at a higher level, but what about the `else` clause here?  That is, we need `value_type_name` to be set to *something* if the `value_type` is an instance of another class.

Reply:
I think this handles every current possibility but I wanted to be very explicit about it, I'm going to change this to throw `ValueError` in the `else` clause to handle any future additions.